File: frontend/models/UserInfoModel.js
import AchievementModel from "./AchievementModel";
import EducationModel from "./EducationModel";
import ExperienceModel from "./ExperienceModel";
import ProjectModel from "./ProjectModel";
import SkillModel from "./SkillModel";

export default class UserInfoModel {
  id = Date.now();
  firstName = "";
  lastName = "";
  email = "";
  phone = "";
  educations = new Array();
  experiences = new Array();
  projects = new Array();
  skills = new Array();
  achievements = new Array();

  constructor(data) {
    if (data != null) {
      this.id = data.id;
      this.firstName = data.firstName;
      this.lastName = data.lastName;
      this.email = data.email;
      this.phone = data.phone;
      this.educations = data.educations.map((item) => {
        return new EducationModel(item);
      });
      this.experiences = data.experiences.map((item) => {
        return new ExperienceModel(item);
      });
      this.projects = data.projects.map((item) => {
        return new ProjectModel(item);
      });
      this.skills = data.skills.map((item) => {
        return new SkillModel(item);
      });
      this.achievements = data.achievements.map((item) => {
        return new AchievementModel(item);
      });
    }
  }
}

File: backend/src/main/java/ResumeBuilder/backend/Models/DatabaseModels/UserInfoModel.java
package ResumeBuilder.backend.Models.DatabaseModels;

import java.util.ArrayList;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonProperty;

import ResumeBuilder.backend.Interfaces.IWriter;
import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;

@Entity
@Table(name = "user_info")
public class UserInfoModel implements IWriter {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    public int id;

    @Column(name = "first_name")
    public String firstName = new String();

    @Column(name = "last_name")
    public String lastName = new String();

    @Column
    public String email = new String();

    @Column
    public String phone = new String();

    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "user_info_id")
    public List<EducationModel> educations = new ArrayList<>();

    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "user_info_id")
    public List<ExperienceModel> experiences = new ArrayList<>();

    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "user_info_id")
    public List<ProjectModel> projects = new ArrayList<>();

    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "user_info_id")
    public List<SkillModel> skills = new ArrayList<>();

    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "user_info_id")
    public List<AchievementModel> achievements = new ArrayList<>();

    public Object get(String key) {
        if (key.equals("name")) {
            return firstName + " " + lastName;
        } else if (key.equals("email")) {
            return email;
        } else if (key.equals("phone")) {
            return phone;
        } else if (key.equals("education")) {
            return educations;
        } else if (key.equals("experience")) {
            return experiences;
        } else if (key.equals("projects")) {
            return projects;
        } else if (key.equals("skills")) {
            return skills;
        } else if (key.equals("achievements")) {
            return achievements;
        }
        return null;
    }
}
